mongo/errors.go

```diff
@@ -227,6 +262,24 @@ func (mwe WriteException) Error() string {
 	return buf.String()
 }
 
+// Unwrap returns nil.
+func (mwe WriteException) Unwrap() error {
+	return nil
+}
```

Comment:
> Still not sure about add Unwrap to the interface when CommandError is the only one that actually uses it.

IIUC, this permits `errors.Is` or `errors.As` on a `ServerError`. It seems harmless to define this for `BulkWriteException` and `WriteException` to return nil.

Apologies for not asking about this in the scope. But if I understand right, `CommandError` only ever wraps the `driver.Error` defined in `x`.

I guess, a `ServerError`, seems to me like it would always be the first error in a chain of errors. It is an externally sourced error, so intuitively we would not wrap it around another error.

If the only thing a `ServerError` can wrap is `driver.Error`, and that is an implementation detail, then maybe we don't want to encourage it by providing an `Unwrap` method?

Reply:
CommandError will wrap whatever the driver.Error that its made from was wrapping. I didn't do an exhaustive check, but I don't think the wrapped error will ever be a driver.Error

Reply:
Ok, I also misinterpreted what `replaceErrors` does. It does not wrap a `driver.Error` directly. It wraps what the `driver.Error` was wrapping.

GODRIVER-1581 gave me a bit more context. `driver.Error` could wrap an underlying network error. That would translate to a `CommandError` with an error label and wrap the original network error.

So I see why `Unwrap` is necessary for a `CommandError

So I guess the question is, do we want to make it convenient for users to check the source error of an underlying `CommandError` using the `ServerError` interface? Alternatively, users can type assert to a `CommandError`. Perhaps @divjotarora or @craiggwilson can chime in.

I suppose it is also odd that `ServerError` could really be a client side network error as well. We add error labels in that case. Should we rename it to `MongoError`?

Reply:
The Unwrap method does not need to be defined on the interface. Simply implementing on the errors that are able to unwrap is enough. No one should be calling Unwrap directly, but only through the use of `errors.Is` or `errors.As`. 

So I'd remove it from the ServerError interface, and not implement it on any custom error type that doesn't wrap another error.

Reply:
done